### PR TITLE
Tx builder/v1

### PR DESCRIPTION
**Pull request by jordanlesich**
## GitHub Issue

https://github.com/HausDAO/daohaus-monorepo/issues/602
https://github.com/HausDAO/daohaus-monorepo/issues/420
https://github.com/HausDAO/daohaus-monorepo/issues/418
https://github.com/HausDAO/daohaus-monorepo/issues/403
 

## Packages Added
 "localforage": "^1.10.0",

## Checks

Before making your PR, please check the following:

- [x] Critical lint errors are resolved
- [x] App runs locally
- [x] App builds locally (run the build command for *any impacted package* and check for any errors before the PR)

### CODE REVIEW

File: apps/core-app/src/pages/FormTest.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+      placeholder: 'Enter title',
+    },
+    {
+      id: 'description',
+      type: 'textarea',
+      label: 'Description',
+      placeholder: 'Enter description',
+    },
+  ],
+  tx: buildMultiCallTX({
```
Comment (skuhlmann) [inline apps/core-app/src/pages/FormTest.tsx:36]: Liking how readable this is

File: apps/core-app/src/pages/Home.tsx [hunk 2]
```diff
@@ -1,48 +1,10 @@
...
-type CustomFields = typeof CustomFields;
-
-const TestForm: FormLego<CustomFields> = {
-  id: 'test',
-  title: 'Test Form',
-  subtitle: 'This is a test form',
-  description: 'This is a test form',
-  fields: [tooltipTest, { type: 'input', id: 'foo' }],
-};
+import { FormBuilder } from '@daohaus/haus-form-builder';
```
Comment (skuhlmann) [inline apps/core-app/src/pages/Home.tsx:2]: cleanup

File: libs/tx-builder-feature/src/utils/cache.ts [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+  '0x2a': {},
+  '0xa': {},
+  '0x64': {},
+  '0x89': {},
+  '0xa4b1': {},
+  '0xa4ec': {},
+};
+
+enum StoreName {
+  ABI = 'ABI',
```
Comment (skuhlmann) [inline libs/tx-builder-feature/src/utils/cache.ts:17]: oh nice - so we can add another if needed.

File: libs/tx-builder-feature/src/utils/search.ts [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+import { processArg } from './args';
+
+export const checkArgType = (arg: unknown) => {
+  if (isArgType(arg)) {
+    return arg;
+  }
+  throw new Error(`Invalid arg type ${arg}`);
+};
+
+export const deepSearch = (
```
Comment (skuhlmann) [inline libs/tx-builder-feature/src/utils/search.ts:18]: i really like the name of this function.

**Review by skuhlmann — APPROVED** [review]

File: libs/tx-builder-feature/src/utils/search.ts (hunk 4 above)
Comment (jordanlesich) [reply libs/tx-builder-feature/src/utils/search.ts:18]: haha

File: libs/tx-builder-feature/src/utils/cache.ts (hunk 3 above)
Comment (jordanlesich) [reply libs/tx-builder-feature/src/utils/cache.ts:17]: Yep. We'll need a few of these.

File: libs/common-utilities/src/types/legoTypes.ts [hunk 5]
```diff
@@ -55,20 +93,40 @@ export type TXLego = {
   finalSuccessMessage?: string;
   errorMessage?: string;
   status?: TxStates;
-  args: ArgAggrageteType;
+  args?: ValidArgType[];
```
Comment (skuhlmann) [inline libs/common-utilities/src/types/legoTypes.ts:96]: Custon spelling in line 91

**Review by skuhlmann — APPROVED** [review]

**Pull request merged**